internal/cloudevents/codec.go

```diff
@@ -10,40 +10,37 @@ import (
 	"github.com/google/uuid"
 	"github.com/kube-orchestra/maestro/internal/db"
 	"k8s.io/apimachinery/pkg/runtime"
-	ceclient "open-cluster-management.io/api/client/cloudevents"
-	"open-cluster-management.io/api/client/cloudevents/types"
+	cegeneric "open-cluster-management.io/api/cloudevents/generic"
+	cetypes "open-cluster-management.io/api/cloudevents/generic/types"
+	workpayload "open-cluster-management.io/api/cloudevents/work/payload"
 )
 
 type Codec struct{}
 
-var _ ceclient.Codec[*db.Resource] = &Codec{}
+var _ cegeneric.Codec[*db.Resource] = &Codec{}
 
-func (codec *Codec) EventDataType() types.CloudEventsDataType {
-	return types.CloudEventsDataType{
-		Group:    "io.open-cluster-management.works",
-		Version:  "v1alpha1",
-		Resource: "manifest",
-	}
+func (codec *Codec) EventDataType() cetypes.CloudEventsDataType {
+	return workpayload.ManifestEventDataType
 }
 
-func (codec *Codec) Encode(source string, eventType types.CloudEventsType, obj *db.Resource) (*cloudevents.Event, error) {
+func (codec *Codec) Encode(source string, eventType cetypes.CloudEventsType, obj *db.Resource) (*cloudevents.Event, error) {
 	evt := cloudevents.NewEvent()
 	evt.SetID(uuid.New().String())
 	evt.SetSource(source)
 	evt.SetType(eventType.String())
 	evt.SetTime(time.Now())
 	evt.SetDataContentType("application/json")
-	evt.SetExtension("resourceID", obj.Id)
-	evt.SetExtension("resourceVersion", obj.ResourceGenerationID)
+	evt.SetExtension("resourceid", obj.Id)
+	evt.SetExtension("resourceversion", obj.ResourceGenerationID)
 	evt.SetExtension("clustername", obj.ConsumerId)
 
 	if !obj.Object.GetDeletionTimestamp().IsZero() {
 		evt.SetExtension("deletionTimestamp", obj.Object.GetDeletionTimestamp().Time)
 		return &evt, nil
 	}
 
-	resourcePayload := &Resource{
-		Manifest: &obj.Object,
+	resourcePayload := &workpayload.Manifest{
+		Manifest: obj.Object,
```

Comment:
Do we always need to set statusfeedback now?

Reply:
The work-client code for testing sets `statusfeedback`.
You're right, we should not relay on work-client to set `statusfeedback`,
I updated encode to set `statusfeedback` explicitly.